clients/horizonclient/error_helpers_test.go

```diff
@@ -82,3 +82,72 @@ func TestIsNotFoundError(t *testing.T) {
 		})
 	}
 }
+
+func TestGetError(t *testing.T) {
+	testCases := []struct {
+		desc    string
+		err     error
+		wantErr error
+	}{
+		{
+			desc:    "nil error",
+			err:     nil,
+			wantErr: nil,
+		},
+		{
+			desc:    "another Go type of error",
+			err:     errors.New("error"),
+			wantErr: nil,
+		},
+		{
+			desc: "not found problem (pointer)",
+			err: &Error{
+				Problem: problem.P{
+					Type:   "https://stellar.org/horizon-errors/not_found",
+					Title:  "Resource Missing",
+					Status: 404,
+				},
+			},
+			wantErr: &Error{
+				Problem: problem.P{
+					Type:   "https://stellar.org/horizon-errors/not_found",
+					Title:  "Resource Missing",
+					Status: 404,
+				},
+			},
+		},
+		{
+			desc: "not found problem (not a pointer)",
+			err: Error{
+				Problem: problem.P{
+					Type:   "https://stellar.org/horizon-errors/not_found",
+					Title:  "Resource Missing",
+					Status: 404,
+				},
+			},
+			wantErr: &Error{
+				Problem: problem.P{
+					Type:   "https://stellar.org/horizon-errors/not_found",
+					Title:  "Resource Missing",
+					Status: 404,
+				},
+			},
+		},
+		{
+			desc:    "a nil *horizonclient.Error",
+			err:     (*Error)(nil),
+			wantErr: nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			gotErr := GetError(tc.err)
+			if tc.wantErr == nil {
+				assert.Nil(t, gotErr)
+			} else {
+				assert.Equal(t, tc.wantErr, gotErr)
+			}
```

Comment:
You can test if gotErr matches wantErr in the case that wantErr is nil, so I don't think if you need the if/else block.
```suggestion
			assert.Equal(t, tc.wantErr, gotErr)
```

If you're having trouble with comparing some error types, you're better off making `wantErr` a `string` of the error output, and then use `assert.EqualError`.

Reply:
https://github.com/stellar/go/pull/2330/files#r384829801